python/lsst/afw/display/saturated.cc

```diff
@@ -0,0 +1,176 @@
+    int const npixMin = 1;              // minimum number of pixels in an object
+    int const SAT = rim.getMask()->getPlaneBitMask("SAT");
+    detection::Threshold const satThresh(SAT, detection::Threshold::BITMASK);
+
+    detection::FootprintSet       sat(*rim.getMask(), satThresh, npixMin);
+    sat.merge(detection::FootprintSet(*gim.getMask(), satThresh, npixMin));
+    sat.merge(detection::FootprintSet(*bim.getMask(), satThresh, npixMin));
+    // go through the list of saturated regions, determining the mean colour of the surrounding pixels
+    typedef detection::FootprintSet::FootprintList FootprintList;
+    PTR(FootprintList) feet = sat.getFootprints();
+    for (FootprintList::iterator ptr = feet->begin(), end = feet->end(); ptr != end; ++ptr) {
+        PTR(detection::Footprint) foot = *ptr;
+        PTR(detection::Footprint) bigFoot = growFootprint(*foot, borderWidth);
+
+        float sumR = 0, sumG = 0, sumB = 0; // sum of all non-saturated adjoining pixels
+        float maxR = 0, maxG = 0, maxB = 0; // maximum of non-saturated adjoining pixels
+
+        for (detection::Footprint::SpanList::const_iterator sptr = bigFoot->getSpans().begin(),
+                 send = bigFoot->getSpans().end(); sptr != send; ++sptr) {
+            PTR(detection::Span) const span = *sptr;
+
+            int const y = span->getY() - y0;
+            if (y < 0 || y >= height) {
+                continue;
+            }
+            int sx0 = span->getX0() - x0;
+            if (sx0 < 0) {
+                sx0 = 0;
+            }
+            int sx1 = span->getX1() - x0;
+            if (sx1 >= width) {
+                sx1 = width - 1;
+            }
+
+            for (typename ImageT::iterator
+                     rptr = rim.at(sx0, y),
+                     rend = rim.at(sx1 + 1, y),
+                     gptr = gim.at(sx0, y),
+                     bptr = bim.at(sx0, y); rptr != rend; ++rptr, ++gptr, ++bptr) {
+                if (!((rptr.mask() | gptr.mask() | bptr.mask()) & SAT)) {
+                    float val = rptr.image();
+                    sumR += val;
+                    if (val > maxR) {
+                        maxR = val;
+                    }
+                    
+                    val = gptr.image();
+                    sumG += val;
+                    if (val > maxG) {
+                        maxG = val;
+                    }
+
+                    val = bptr.image();
+                    sumB += val;
+                    if (val > maxB) {
+                        maxB = val;
+                    }
+                }
+            }
+        }
+        // OK, we have the mean fluxes for the pixels surrounding this set of saturated pixels
+        // so we can figure out the proper values to use for the saturated ones
+        float R = 0, G = 0, B = 0;      // mean intensities
+        if(sumR + sumB + sumG > 0) {
+            float scale = 1.0;          // how much to scale up max values
```

Comment:
An experiment to make the cores of brighter, but it's not configurable and I've removed it.